### DISCUSSION TITLE
PredictPose slow frame rate

### DISCUSSION THREAD

**Discussion by ZacariasBarreto** [bonsai-rx/bonsai#1807] [Q&A] [answered]
Hi,

I am using bonsai with DLC, using PredictPose after a FileCapture node; and, although when I use only the FileCapture node, the video plays at its usual frame rate, when I join the PredictPose node, the frame rate drops significantly, to the point where it would be useless in real time if it would take this much time to analyze each frame (I am using it with FileCapture just to get everything ready to switch to CameraCapture, when the rest of the experiment's setup is ready).  Do you have any idea how I could mitigate this?

Thanks in advance,
zbx

**Comment by bruno-f-cruz** [comment, marked as answer]
Seems like your machine can't run the inference as fast as the video playbacks. Assuming you are using a gpu to run it, you can use a better machine or try to reduce the number of frames per second using the Slice operator for instance.

**Reply by ZacariasBarreto** [reply]
Hi,
Unfortunately, I am using a computer that doesn't have a GPU, because I'm an intern, and that's what they had available for me.
If I use Slice, it will probably still take the same time for each frame, won't it? I have a feeling the problem will stay the same.
The current ideas to get rid of this problem (apart from getting a computer with a GPU) are:
1. Using a Buffer that will only show for instance 30 frames when they are ready, and these would be shown at a normal fps rate, but, when I use a Buffer, I can't get the position of specific labels out of it, which I need;
2. Drop the number of labels the system predicts - is this possible to do in Bonsai DLC?

**Reply by bruno-f-cruz** [reply]
Using the Buffer will have the same problem as Slice. The time it takes to push one frame is the same. You won't be able to do real time at full frame rate.

Regarding number of labels, you can't drop it in bonsai since the network is frozen. Even if you could the gains are usually very small.

**Comment by ZacariasBarreto** [comment]
Hi,

Buffer will take the same time to push one frame but if I just want to simulate what could happen in a better computer, could it label like 30 frames across a certain amount of time, passing each frame to the buffer, and, when they are all ready, those labeled frames play at normal speed ? It would then stop again and wait for the next group of labeled frames. I just don't know how to make it show these frames at a normal speed when they are all ready, and get the positions of each label from the Buffer. 

Do you reckon the only solution would be using a better computer?

Thanks,

**Reply by bruno-f-cruz** [reply]
Not necessarily a better computer. A GPU to run inference will make a big difference already.

**Reply by ZacariasBarreto** [reply]
thanks. It did
